pandas/_libs/tslibs/parsing.pyx

```diff
@@ -709,8 +707,8 @@ class _timelex:
             self.stream = instream
         elif getattr(instream, 'read', None) is None:
             raise TypeError(
-                'Parser must be a string or character stream, not '
-                '{itype}'.format(itype=instream.__class__.__name__))
+                f'Parser must be a string or character stream, not '
+                f'{instream.__class__.__name__}')
```

Comment:
can you change this to `type(instream)`?  this is another thing we're not too consistent about (also type(foo) vs type(foo).__name__) and help improving that would be enthusiastically welcomed

Reply:
sure, will create a separate PR that will cover all of those. for now Ill fix just that